hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/heartbeat/HoodieHeartbeatClient.java

```diff
@@ -228,7 +228,10 @@ public boolean isHeartbeatExpired(String instantTime) throws IOException {
       lastHeartbeatForWriter.setInstantTime(instantTime);
       lastHeartbeatForWriter.getTimer().cancel();
     }
-    if (currentTime - lastHeartbeatForWriter.getLastHeartbeatTime() > this.maxAllowableHeartbeatIntervalInMs) {
+    // When the last write had failed, the heartbeat client would close with stopping all heartbeat which
+    // includes deleting heartbeat file. Therefore, it isn't heartbeat expired for the last failed write.
+    if (lastHeartbeatForWriter.getLastHeartbeatTime() != 0L
+        && currentTime - lastHeartbeatForWriter.getLastHeartbeatTime() > this.maxAllowableHeartbeatIntervalInMs) {
       LOG.warn("Heartbeat expired, currentTime = " + currentTime + ", last heartbeat = " + lastHeartbeatForWriter
```

Comment:
There maybe some use cases that some stray tasks are still there and we want to commit the instant, but in general, fix it would avoid some error reportings like the instant recommit.